.github/workflows/test.yml

```
@@ -32,229 +35,24 @@ jobs:
-          kbuild-output: ${{ env.KBUILD_OUTPUT }}
-          max-make-jobs: 32
-          llvm-version: ${{ matrix.llvm-version }}
-      - name: Tar artifacts
-        run: |
-          bash .github/scripts/tar-artifact.sh ${{ matrix.arch }} ${{ matrix.toolchain_full }}
-      - if: ${{ github.event_name != 'push' }}
-        name: Remove KBUILD_OUTPUT content
-        shell: bash
-        run: |
-          # Remove $KBUILD_OUTPUT to prevent cache creation for pull requests.
-          # Only on pushed changes are build artifacts actually cached, because
-          # of github.com/actions/cache's cache isolation logic.
-          rm -rf "${KBUILD_OUTPUT}"
-      - uses: actions/upload-artifact@v3
-        with:
-          name: vmlinux-${{ matrix.arch }}-${{ matrix.toolchain_full }}
-          if-no-files-found: error
-          path: vmlinux-${{ matrix.arch }}-${{ matrix.toolchain_full }}.tar.zst
-  test:
-    if: ${{ github.event_name != 'push' }}
-    name: ${{ matrix.test }} on ${{ matrix.arch }} with ${{ matrix.toolchain_full }}
-    needs: [set-matrix, build]
-    strategy:
-      fail-fast: false
-      matrix: ${{ fromJSON(needs.set-matrix.outputs.test-matrix) }}
-    runs-on: ${{ matrix.runs_on }}
-    timeout-minutes: 100
-    env:
-      KERNEL: ${{ matrix.kernel }}
-      REPO_ROOT: ${{ github.workspace }}
-      REPO_PATH: ""
-      KBUILD_OUTPUT: kbuild-output/
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/download-artifact@v3
-        with:
-          name: vmlinux-${{ matrix.arch }}-${{ matrix.toolchain_full }}
-          path: .
-      - name: Untar artifacts
-        # zstd is installed by default in the runner images.
-        run: zstd -d -T0  vmlinux-${{ matrix.arch }}-${{ matrix.toolchain_full }}.tar.zst --stdout | tar -xf -
-      - name: Prepare rootfs
-        uses: libbpf/ci/prepare-rootfs@main
-        with:
-          project-name: 'libbpf'
-          arch: ${{ matrix.arch }}
-          kernel: ${{ matrix.kernel }}
-          kernel-root: '.'
-          kbuild-output: ${{ env.KBUILD_OUTPUT }}
-          image-output: '/tmp/root.img'
-          test: ${{ matrix.test }}
-      - name: Run selftests
-        uses: libbpf/ci/run-qemu@main
-        continue-on-error: ${{ matrix.continue_on_error }}
-        timeout-minutes: ${{ matrix.timeout_minutes }}
-        with:
-          arch: ${{ matrix.arch}}
-          img: '/tmp/root.img'
-          vmlinuz: '${{ github.workspace }}/vmlinuz'
-          kernel-root: '.'
-          max-cpu: 8
-          kernel-test: ${{ matrix.test }}
-  veristat:
-    name: veristat
-    needs: [set-matrix, build]
-    runs-on: ${{ fromJSON(needs.set-matrix.outputs.veristat-runs-on) }}
-    timeout-minutes: 100
-    env:
-      KERNEL: LATEST
-      REPO_ROOT: ${{ github.workspace }}
-      REPO_PATH: ""
-      KBUILD_OUTPUT: kbuild-output/
+
+  build-and-test:
+    uses: ./.github/workflows/kernel-build-test.yml
+    needs: [set-matrix]
```

```suggestion
  build-and-test:
    name: "{{ matrix.arch }}-{{ matrix.toolchain_full }}"
    uses: ./.github/workflows/kernel-build-test.yml
    needs: [set-matrix]
```

It would render job name in a nicer way

---

done! thanks